guardian/editions
# add real weather location consent, fetching and display

## Summary

We want to [have weather use people's real location](https://trello.com/c/Ii5WqyrA/895-make-weather-in-app-use-geolocation-data-lat-long-rather-than-ip-address) when they agree to give us location information.

This changeset adds a new consent screen, which gets open when people press "Set Location" on the weather widget. That button shows up only if we don't have the permission yet. If we get the permission, we show the real location.

This changeset further leverage Apollo, as its dependency #760, to manage data easily. If we decide not to go forward with Apollo, I'd replace it with custom data management (I don't recommend it: more boilerplate, more bugs, etc.)

This changeset might be a big too large for comfortable review. Please let me know if you think so, and I can consider ways to split that up a little bit!

## Test Plan

1. Open fresly installed app with empty system cache.
2. Notice weather shows up, with a "Set Location" button.
3. Press button. Observe modal pops up explaining how we use geolocation.
4. Press "I understand". Observe the OS's dialog showing up asking you for permission.
5. Deny the permission. Observe it just comes back to the explanation.
6. Press "I understand". Press "Deny and don't ask again" (on Android at least). Notice the consent modal is dismissed, and weather shows IP-based location.
7. Go to app system settings, reset permission. Switch back to app.
8. Press "Set Location", then "I understand". Observe the modal is dismissed, then the correct weather location and forecasts show up.
9. Go to app system settings, reset permission. Switch back to app. Observe it shows "Set Location" again, which proves state is cleanly and properly handled.

<div>
<img width="200" alt="Screenshot 2019-10-25 at 15 53 05" src="https://user-images.githubusercontent.com/1733570/67581224-91a4ae80-f73f-11e9-99ac-1ba9b7b5f1af.png">
<img width="200" alt="Screenshot 2019-10-25 at 15 46 39" src="https://user-images.githubusercontent.com/1733570/67581226-91a4ae80-f73f-11e9-806b-995e762f58de.png">
<img width="200" alt="Screenshot 2019-10-25 at 15 55 47" src="https://user-images.githubusercontent.com/1733570/67581491-037cf800-f740-11e9-84da-48f259a22d28.png">
</div>
